Shared/Samples/List geodatabase versions/ListGeodatabaseVersionsView.swift

```diff
@@ -0,0 +1,133 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ListGeodatabaseVersionsView: View {
+    /// The analysis error shown in the error alert.
+    @State private var analysisError: Error?
+    /// The task used to perform geoprocessing jobs.
+    @State private var geoprocessingTask = GeoprocessingTask(
+        url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/GDBVersions/GPServer/ListVersions")!
+    )
+    /// The geodatabase version information.
+    @State private var geodatabaseVersionInfos: [GeodatabaseVersionInfo]?
+    
+    var body: some View {
+        Form {
+            if let geodatabaseVersionInfos {
+                ForEach(geodatabaseVersionInfos, id: \.objectID) { version in
+                    Section("OBJECTID: \(version.objectID)") {
+                        LabeledContent("Name", value: version.name)
+                            .lineLimit(1)
+                            .textSelection(.enabled)
+                        LabeledContent("Access", value: version.access)
+                        LabeledContent("Created", value: version.created.formatted(date: .abbreviated, time: .shortened))
+                        LabeledContent("Last Modified", value: version.lastModified.formatted(date: .abbreviated, time: .shortened))
+                        LabeledContent("Is Owner", value: version.isOwner)
+                        LabeledContent("Parent Version", value: version.parentVersionName)
+                        if !version.description.isEmpty {
+                            LabeledContent("Description", value: version.description)
+                        }
+                    }
+                }
+            } else {
+                ProgressView()
+                    .frame(maxWidth: .infinity)
+            }
+        }
+        .task {
+            do {
+                if let resultFeatures = try await fetchGeodatabaseVersions() {
+                    geodatabaseVersionInfos = makeGeodatabaseVersionInfos(from: resultFeatures)
+                }
+            } catch {
+                analysisError = error
+            }
+        }
+        .errorAlert(presentingError: $analysisError)
+    }
+    
+    /// Fetches geodatabase versions using a geoprocessing job.
+    private func fetchGeodatabaseVersions() async throws -> FeatureSet? {
```

Comment:
Throwing and also returning an optional might not always be the best choice. Should you throw some well-defined error instead of returning `nil` or does an optional really make sense here?

Reply:
From core yaml:

> (GeoprocessingFeatures.features) It can be nil for services that have a map service, in which case calling `fetchOutputFeatures()` will retrieve the features from the service.

Thus, as long as the service provided supports versioning, the output `featureSet` won't be `nil` if the geoprocessing job successfully finishes with the results. I decided to `fatalError` in the case of `nil`.